common/docker/client.go

```diff
@@ -157,14 +157,35 @@ func (pending pendingStarts) finish(id string) {
 }
 
 func (c *Client) AllContainerIDs() ([]string, error) {
-	all, err := c.ListContainers(docker.ListContainersOptions{All: true})
+	return c.containerIDs()
+}
+
+// ExistingContainerIDs returns a list of container IDs which either are running or
+// might be started in the future.
+func (c *Client) ExistingContainerIDs() ([]string, error) {
+	return c.containerIDs("running", "paused", "exited", "created", "restarting")
```

Comment:
OK, thanks.  Also https://github.com/docker/docker/blob/master/daemon/start.go#L98

Is it worth making this distinction?  Does it do anything apart from let us remove tombstoned entries earlier?

Having an inclusive list like this is fragile - it will break if the set of states is extended.